### PR TITLE
feat: add documentation for mcp auto test feature

### PR DESCRIPTION
**Pull request by yogeshnikam671**

### CODE REVIEW

File: documentation/mcp_auto_test.md [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+title: MCP Auto Test
+parent: Documentation
+nav_order: 6
+---
+
+# MCP Auto Test
+
+## Automated Regression Testing for MCP Tools  
+MCP Auto Test is a robust tool designed to automate the testing of Model Context Protocol (MCP) tools exposed by an MCP Server. Unlike existing utilities such as MCP Inspector—which is limited to manual, ad hoc testing—MCP Auto Test provides a reproducible, fully automated framework for regression, edge case, and input-combination testing. This ensures greater reliability and faster release cycles as new features are added to the MCP ecosystem.
+
```
Comment (yogeshnikam671) [inline documentation/mcp_auto_test.md:14]: Start with "Why" instead with "What".
Specify the problem statement first and then explain where auto test comes in.

File: documentation/mcp_auto_test.md [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+- Detect regressions immediately as the MCP server or its tools evolve.
+
+By integrating MCP Auto Test, teams can accelerate their delivery pipelines and confidently make frequent changes to MCP tools without introducing risk.
+
+---
+
+## Key Features
+
+- **Automated Test Generation**: Dynamically builds and runs test cases for all defined MCP tools, leveraging schemas and user-provided sample data.
+- **Comprehensive Input Coverage**: Explores input permutations, including edge cases, to validate tool robustness beyond the "happy path."
```
Comment (yogeshnikam671) [inline documentation/mcp_auto_test.md]: Say something on the lines of "wide range of tests generated based on your tool schema".

File: documentation/mcp_auto_test.md [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+- **Edge Case & Resiliency Detection**: Simulates both valid and invalid argument combinations, surfacing issues that might only appear in production.
+- **Regression Prevention**: Allows rapid reruns of the complete test suite after any change, protecting against accidental breaking changes or feature regressions.
```
Comment (yogeshnikam671) [inline documentation/mcp_auto_test.md]: Avoid regression

File: documentation/mcp_auto_test.md [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+- **Flexible Integration**: Can be invoked with Docker in CI/CD or local workflows, and supports optional bearer authentication.
```
Comment (yogeshnikam671) [inline documentation/mcp_auto_test.md]: Runs on local or CI

File: documentation/mcp_auto_test.md [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+- **Structured Reporting**: Outputs both tool schemas and JSON test reports for auditability and debugging.
```
Comment (yogeshnikam671) [inline documentation/mcp_auto_test.md]: Detailed test report

File: documentation/mcp_auto_test.md [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
+**Tip:** You can provide diverse or minimal datasets as required. The tool's runner will internally generate a variety of tests leveraging this dictionary.
+
+---
+
+### 2. Running Automated Tests
+
+Once your dictionary file is ready, launch MCP Auto Test against your target MCP server with a single Docker command:
+
+```bash
+docker run -v "$(pwd)/build/reports/specmatic:/usr/src/app/build/reports/specmatic" \
```
Comment (yogeshnikam671) [inline documentation/mcp_auto_test.md:182]: add the dictionary volume mounting

File: documentation/mcp_auto_test.md [hunk 7]
```diff
@@ -0,0 +1,448 @@
...
+
+### 3. Enabling Resiliency Testing
+
+MCP Auto Test provides optional **resiliency** scenarios—thoroughly exercising input validation and uncovering boundary bugs or regressions.
+
+To enable resiliency testing:
+
+```bash
+docker run -v "$(pwd)/build/reports/specmatic:/usr/src/app/build/reports/specmatic" \
+  specmatic/specmatic mcp test \
```
Comment (yogeshnikam671) [inline documentation/mcp_auto_test.md:218]: add volume mount for the dictionary file
